### PR TITLE
Default variant validation on SetState

### PR DESCRIPTION
**Pull request by skyerus**

### CODE REVIEW

**Comment by beeme1mr** [comment]
Please add a test for this.

**Review by AlexsJones — APPROVED** [review]

**Comment by AlexsJones** [comment]
Thank you for your contribution @skyerus

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -18,10 +19,13 @@ var (
...
-		return err
+		return fmt.Errorf("fetch: %w", err)
 	}
 	mu.Lock()
-	_ = ev.SetState(msg)
+	err = ev.SetState(msg)
+	if err != nil {
+		return fmt.Errorf("set state: %w", err)
+	}
 	mu.Unlock()
```
Comment (AlexsJones) [inline pkg/runtime/runtime.go:29]: This mutex will never unlock, can we defer it before the set state?

Comment (AlexsJones) [reply pkg/runtime/runtime.go:29]: @skyerus ^

Comment (skyerus) [reply pkg/runtime/runtime.go]: Nice spot, done. I'm surprised this wasn't picked up by the linter

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**